### PR TITLE
[Main Course] - Flaskbb2.0

### PR DESCRIPTION
**Pull request by jhqian0909**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Post for Flaskbb2.0

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Tina0330 — APPROVED** [review]
LGTM,thanks

**Comment by EchoTang04** [comment]
提个问题，有预览链接吗？还是直接看文字内容就好？

**Comment by jhqian0909** [comment]
> 提个问题，有预览链接吗？还是直接看文字内容就好？

https://github.com/BulletTech/BulletTech/blob/flaskbb/docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md

File: docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+---
+
+# 解读FlaskBB源码
+
+!!! info
+    作者：Void，发布于2021-10-29，阅读时间：约6分钟，微信公众号文章链接：[:fontawesome-solid-link:](https://mp.weixin.qq.com/s/Bsoc0rj14ma3luKjvR_9qQ)
+
+## 1 前言
+
+上回说到FlaskBB是一个开源的功能强大的论坛，我们也已经成功地在本地运行了起来。不止于此，本文将带着大家解读FlaskBB的源码。
```
Comment (Realvincentyuan) [inline docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md:12]: 可以引用上一篇 ：）

File: docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+## 2 目录结构
+
+我们首先来看看它的目录结构。在根目录下，可以看到有常见的依赖相关的文件(requirements.txt，setup.py等)，也有刚刚通过flaskbb makeconfig生成的flaskbb.cfg。tests文件夹存放了自动测试相关的代码。对于一个产品级的应用，自动测试是不可或缺的。
```
Comment (Realvincentyuan) [inline docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md:16]: 这里是少了一个文件树吗？哪里有目录结构？

File: docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+我们首先来看看它的目录结构。在根目录下，可以看到有常见的依赖相关的文件(requirements.txt，setup.py等)，也有刚刚通过flaskbb makeconfig生成的flaskbb.cfg。tests文件夹存放了自动测试相关的代码。对于一个产品级的应用，自动测试是不可或缺的。  
+最后是主文件夹flaskbb。
+
+## 3 源码解读
+
+我们来看看flaskbb的主程序app.py中的create_app函数：
+
+```python
+    configure_app(app, config)
```
Comment (Realvincentyuan) [inline docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md:24]: 这里是少了函数头吗？参数和返回值没看到

**Review by Realvincentyuan — COMMENTED** [review]
是不是少放了一些截图和函数信息啊？感觉看起来很抽象 😂

File: docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md (hunk 1 above)
Comment (jhqian0909) [reply docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md]: 已引用

File: docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md (hunk 2 above)
Comment (jhqian0909) [reply docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md]: 已更新

File: docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md (hunk 3 above)
Comment (jhqian0909) [reply docs/Main_Course/One_Day_Build/2021-10-29-flaskbb2.md:40]: 已更新

**Review by jz263 — APPROVED** [review]

**Pull request merged**